docs/release-strategy.md

```diff
@@ -0,0 +1,68 @@
+# InstructLab CLI Release Strategy
+
+This document discusses the release strategy and processes for the
+`instructlab` Python package built from the
+<https://github.com/instructlab/instructlab> git repository.
+
+## Versioning Scheme
+
+Releases use a `X.Y.Z` numbering scheme.
+
+X-stream release are for major releases. At this stage in the project a major release has not been cut and we expect each release (2-3 weeks) to be a new Y-stream.
+
+Backwards compatibility will exist within a Y-stream for example within 0.1.8 and 0.1.9 and between Y-stream releases for update purposes. For example between 0.1.9 and 0.2.0.
+
+Z-stream releases are meant for critical bug and documentation fixes. Backward compatibility is guaranteed between Z-stream releases. Z-stream releases are cut as maintainers see fit.
+
+## Schedule
+
+The project currently operates on a time-based release schedule.
+The goal is to release a new Y-stream every 2-3 weeks though the time span remains flexible based on the discretion of the maintainers team.
+
+The cadence for major releases starting from 1.0 and onward will be determined as the project matures.
+
+A schedule will be updated in a markdown file on the <https://github.com/instructlab/instructlab> GitHub repository.
+
+## Release Tracking
+
+Release planning is tracked via milestones in GitHub. Milestones on GitHub will exist for the next two releases.
+
+PRs and issues associated with the next two milestones will be prioritized for review and merging into the main branch of instructlab.
+
+## Git Branches and Tags
+
+Every `X.Y` release stream gets a new branch.
+
+Each release, `X.Y.Z`, exists as a tag named `vX.Y.Z` on the `release-vX.Y` branch.
```

Comment:
IMHO, we should delete all branches except for `main` and `release-0.14` currently present - none of the existing release branches follow this scheme and it's confusing frankly